src/main.cpp

```diff
@@ -155,11 +155,17 @@ int main(int argc, char *argv[])
 #else
     wlr_log_init(WLR_ERROR, NULL);
 #endif
+    const char *config_basename = "wayfire.ini";
 
     std::string config_dir = nonull(getenv("XDG_CONFIG_DIR"));
     if (!config_dir.compare("nil"))
         config_dir = std::string(nonull(getenv("HOME"))) + "/.config/";
-    config_file = config_dir + "wayfire.ini";
+    config_file = config_dir + config_basename;
+
+    /* use system-wide config file if local is not accessible */
+    /* FIXME: there can be a race */
```

Comment:
Can you elaborate on this race potential here?

Reply:
This can happen in case when an user simultaneously run Wayfire and erases the config file. The right thing to do is to fallback to system config file when Wayfire fails to open user config.

Reply:
I think this is fine, if the config file is erased, we just stop reloading changes dynamically (at least in theory).